src/synthesizer/particle/stars.py

```diff
@@ -1605,6 +1605,81 @@ def get_particle_spectra_reprocessed(
 
         return reprocessed
 
+    def get_particle_spectra_screen(
+        self,
+        grid=None,
+        fesc=0.0,
+        tau_v=None,
+        dust_curve=PowerLaw(slope=-1.0),
+        mask=None,
+        method="cic",
+        label="",
+    ):
+        """
+        Generates the dust attenuated spectra. First generates the intrinsic
+        spectra if this hasn't already been calculated.
+
+        Args:
+            grid (obj):
+                Spectral grid object.
+            fesc (float/array-like, float)
+                Fraction of stellar emission that escapes unattenuated from
+                the birth cloud. Can either be a single value
+                or an value per star (defaults to 0.0).
+            kwargs
+                Any keyword arguments which can be passed to
+                generate_particle_lnu.
+
+        Updates:
+            incident:
+            transmitted
+            nebular
+            reprocessed
+            intrinsic
+            attenuated
+
+            if fesc>0:
+                escaped
+
+        Returns:
+            Sed
+                An Sed object containing the attenuated spectra.
+        """
+
```

Comment:
Need an 
```
if label[-1] != "_":
    label = f"{label}_"
```

here.

Reply:
Also need to create a True everywhere mask `if mask is None`

Reply:
Hmm... do we really want the label thing? Won't that fail is label=""?

Reply:
Ah my bad:
```
if len(label) > 0 and label[-1] != "_":
    label = f"{label}_"
```

Reply:
I appreciate it's nice to enforce labels include an underscore, but don't we need to add this all over the place.

Reply:
It's already handled in the integrated stuff that takes a label

Reply:
Won't it need adding in every method that takes a label?

Reply:
OK, done I think